### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by carmocca**
Generation works.

I removed all files that are not updated for simplicity. We can port them from upstream on demand.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,174 +1 @@
...
-- [@karpathy](https://github.com/karpathy) for [nanoGPT](https://github.com/karpathy/nanoGPT)
-- [@FacebookResearch](https://github.com/facebookresearch) for the original [LLaMA implementation](https://github.com/facebookresearch/llama)
-- [@TimDettmers](https://github.com/TimDettmers) for [bitsandbytes](https://github.com/TimDettmers/bitsandbytes)
-- [@Microsoft](https://github.com/microsoft) for [LoRA](https://github.com/microsoft/LoRA)
-- [@IST-DASLab](https://github.com/IST-DASLab) for [GPTQ](https://github.com/IST-DASLab/gptq)
-
-## License
-
-Lit-LLaMA is released under the [Apache 2.0](https://github.com/Lightning-AI/lightning-llama/blob/main/LICENSE) license.
+# FIXME
```
Comment (carmocca) [inline README.md:1]: We probably want to refresh this

File: howto/inference.md [hunk 2]
```diff
@@ -15,6 +15,8 @@ The script assumes you have downloaded and converted the weights and saved them
 > **Note**
 > All scripts support argument [customization](customize_paths.md)
 
+### FIXME: update this
```
Comment (carmocca) [inline howto/inference.md:18]: Need to try this on a A100

File: lit_stablelm/model.py [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+        cos, sin = cos[:T], sin[:T]
+
+        q_roped = apply_rope(q[..., :n_elem], cos, sin)
+        k_roped = apply_rope(k[..., :n_elem], cos, sin)
+        q = torch.cat((q_roped, q[..., n_elem:]), dim=-1)
+        k = torch.cat((k_roped, k[..., n_elem:]), dim=-1)
+
+        if hasattr(self, "bias"):
+            # causal self-attention; Self-attend: (B, nh, T, hs) x (B, nh, hs, T) -> (B, nh, T, T)
+            # NOTE: cannot use flash attention because it takes q.size(-1) as the norm factor which is different to the
```
Comment (lantiga) [inline lit_stablelm/model.py:156]: why is this conditioned on bias being there?

File: lit_stablelm/tokenizer.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+from pathlib import Path
+from typing import Optional
+
+import torch
+from tokenizers import Tokenizer as HFTokenizer
+
+
+class Tokenizer:
+    def __init__(self, vocabulary_path: Path, config_path: Path) -> None:
+        # https://github.com/Stability-AI/StableLM/blob/e60081/configs/stablelm-base-alpha-3b.yaml#L108
```
Comment (lantiga) [inline lit_stablelm/tokenizer.py:11]: I think we should just vendor the yaml file in the repo directly

**Review by lantiga — APPROVED** [review]
Fantastic work @carmocca!

**Comment by lantiga** [comment]
There's a test failing in windows and the readme to complete. I can work on the readme.

**Pull request merged**